### ISSUE TITLE
Gradle fails to resolve dependencies with ivy metadata with recursive defaultConfMappings

### ISSUE THREAD

**Issue by big-guy** [gradle/gradle#9842]
Hacky PR:
https://github.com/gradle/gradle/pull/9822

**Comment by ljacomet** [comment]
So the recursive aspect is about the [fallback mechanism](https://ant.apache.org/ivy/history/latest-milestone/ivyfile/dependency.html#_specify_a_configuration_mapping_using_the_code_code_operator) of Ivy.

So with something like `compile->compile(compile(*))`, Gradle fails to resolve as it tries to find `compile(*)` which does not exist.

However Ivy, fails as well. Since it uses a [regexp](https://github.com/apache/ant-ivy/blob/0d07c9985cfaa96172a154f4eeea2f2c348f8c3f/src/java/org/apache/ivy/core/resolve/IvyNode.java#L70) to extract the fallback, it will try to resolve `*)` which is invalid as well.

In conclusion, I do not think that Gradle should support that use case as it simply is a case of invalid metadata.

**Comment by ghale** [comment]
Where does this stand?  Is there anything to do here or are we chalking this up to bad metadata?

**Comment by ljacomet** [comment]
Closing this as it is related to bad metadata.
